test/ffi/booster.jl

```diff
@@ -403,6 +403,33 @@ end
 end
 
 
+@testset "LGBM_BoosterRefit" begin
+    # Sample dataset
+    X_train = randn(1000, 20)
+    y_train = rand([0, 1], 1000)
+    
+    # Create an estimator with predict_leaf_index set to true to obtain leaf index predictions
+    estimator = LightGBM.LGBMClassification(objective = "binary", num_class = 1, predict_leaf_index = true, num_iterations = 20)
+    
+    # Fit the estimator with the training data
+    LightGBM.fit!(estimator, X_train, y_train, verbosity = -1)
+    
+    # Get the leaf predictions using the training data
+    leaf_predictions = LightGBM.predict(estimator, X_train)
+
+    # Convert to Int32 which is expected by the C API and should be for leaf indices
+    # Reshape the leaf predictions as unlike the normal or raw predictions their number of rows is number of data points * num_trees/iterations 
+    reshaped_leaf_predictions = reshape(convert(Matrix{Int32}, leaf_predictions), (size(X_train, 1), estimator.num_iterations))
```

Comment:
This conversion can be probably done on predict level if parameter is predict_leaf_index because indices are not floats. As for reshaping it could be an optional param in predict. Reshaping can also be done prior to passing over to LGBM_BoosterRefit as that's where it's needed.